src/cdk/testing/testbed/unit-test-element.ts

```diff
@@ -126,9 +126,16 @@ export class UnitTestElement implements TestElement {
     await this._stabilize();
   }
 
-  async text(): Promise<string> {
-    await this._stabilize();
-    return (this.element.textContent || '').trim();
+  async text(options?: {excludes?: string}): Promise<string> {
+    await this._stabilize();
+    if (options?.excludes) {
+      const clone = this.element.cloneNode(true);
+      const exclusions = this.element.querySelectorAll(options.excludes);
+      exclusions.forEach(n => n.remove());
```

Comment:
A few browser compatibility issues here:
1. `querySelectorAll` returns a `NodeList` which doesn't have a `forEach` method on all browsers. This should be changed to a plain `for` loop.
2. `remove` isn't supported in all browsers, it should be changed to `n.parentNode.removeChild(n)`.